### PR TITLE
feat(wallet): New Quote Countdown

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Introduces a `countdown` ticker showing the time in until a new `quote` is fetched on the `Swap` and `Bridge` screens.

Resolves <https://github.com/brave/brave-browser/issues/34583>

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Open the `Wallet` and navigate to the `Swap` or `Bridge` screen
2. Generate a `quote` for a `swap`.
3. You should see a `countdown` ticker displayed, showing the time until a new `quote` is fetched.

https://github.com/brave/brave-core/assets/40611140/af9c43e9-df4f-4e63-bcb5-b182486dbd37

### CODE REVIEW

**Review by yrliou — APPROVED** [review]
core++

File: components/resources/wallet_strings.grdp [hunk 1]
```diff
@@ -924,10 +924,11 @@
   <message name="IDS_BRAVE_SWAP_HALF" desc="Amount preset button text for half">Half</message>
   <message name="IDS_BRAVE_SWAP_MAX" desc="Amount preset button text for max">Max</message>
   <message name="IDS_BRAVE_SWAP_ENTER_AN_AMOUNT" desc="Text prompting to enter an amount">Enter an amount</message>
-  <message name="IDS_BRAVE_SWAP_FINDING_PRICE" desc="Loader text while quotes are fetched">Finding the best quote...</message>
+  <message name="IDS_BRAVE_WALLET_FETCHING_QUOTE" desc="Loader text while quotes are fetched">Fetching quote</message>
   <message name="IDS_BRAVE_WALLET_RECEIVE_ESTIMATE" desc="Swap receive estimate label">Receive (est.)</message>
   <message name="IDS_BRAVE_WALLET_FROM_TOKEN" desc="Swap from token label">From token</message>
   <message name="IDS_BRAVE_WALLET_TO_TOKEN" desc="Swap to token label">To token</message>
+  <message name="IDS_BRAVE_WALLET_NEW_QUOTE_IN" desc="Swap new quote in label">New quote in <ph name="COUNTDOWN_BEFORE">$1</ph>$countdown<ph name="COUNTDOWN_AFTER">$2</ph></message>
```
Comment (mkarolin) [inline components/resources/wallet_strings.grdp:931]: Seems like countdown is also a placeholder? Would be easier for the localizers to understand if it were: `<ph name="COUNTDOWN_BEFORE">$1</ph><ph name="COUNTDOWN">$1<ex>00:05</ex></ph><ph name="COUNTDOWN_AFTER">$2</ph>`. If this makes things too complicated on React side, then at least let's add something to the `desc` that explains that `$countdown` would be replaced with a digital counting down clock or such.

Comment (Douglashdaniel) [reply components/resources/wallet_strings.grdp]: @mkarolin good point, I have updated this with a the `<ex></ex>` tags! https://github.com/brave/brave-core/compare/47906f3f4a9e06a43ecd6b666fa27302ebe50b39..f7de84c2d9b6178d713dd1e7e3f63d323cb513ed

**Review by mkarolin — APPROVED** [review]
`strings` LGTM

**Pull request merged**